[2025-04-13T21:58:41.948+0800] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-13T21:58:41.952+0800] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: reddit_etl_pipeline.extract_reddit_data manual__2025-04-13T13:22:15.449641+00:00 [queued]>
[2025-04-13T21:58:41.955+0800] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: reddit_etl_pipeline.extract_reddit_data manual__2025-04-13T13:22:15.449641+00:00 [queued]>
[2025-04-13T21:58:41.955+0800] {taskinstance.py:2867} INFO - Starting attempt 2 of 2
[2025-04-13T21:58:41.968+0800] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): extract_reddit_data> on 2025-04-13 13:22:15.449641+00:00
[2025-04-13T21:58:41.971+0800] {standard_task_runner.py:72} INFO - Started process 62503 to run task
[2025-04-13T21:58:41.973+0800] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'reddit_etl_pipeline', 'extract_reddit_data', 'manual__2025-04-13T13:22:15.449641+00:00', '--job-id', '14', '--raw', '--subdir', 'DAGS_FOLDER/reddit_dag.py', '--cfg-path', '/tmp/tmpj3i21ra2']
[2025-04-13T21:58:41.974+0800] {standard_task_runner.py:105} INFO - Job 14: Subtask extract_reddit_data
[2025-04-13T21:58:42.000+0800] {task_command.py:467} INFO - Running <TaskInstance: reddit_etl_pipeline.extract_reddit_data manual__2025-04-13T13:22:15.449641+00:00 [running]> on host Conglaptop.
[2025-04-13T21:58:42.036+0800] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='reddit_etl_pipeline' AIRFLOW_CTX_TASK_ID='extract_reddit_data' AIRFLOW_CTX_EXECUTION_DATE='2025-04-13T13:22:15.449641+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-13T13:22:15.449641+00:00'
[2025-04-13T21:58:42.037+0800] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-13T21:58:42.037+0800] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-13T21:58:42.037+0800] {logging_mixin.py:190} INFO - Current task name:extract_reddit_data state:running start_date:2025-04-13 13:58:41.952979+00:00
[2025-04-13T21:58:42.037+0800] {logging_mixin.py:190} INFO - Dag name:reddit_etl_pipeline and current dag run status:running
[2025-04-13T21:58:42.037+0800] {taskinstance.py:732} INFO - ::endgroup::
[2025-04-13T21:58:42.037+0800] {reddit_dag.py:54} INFO - Starting Reddit data extraction
[2025-04-13T21:58:43.252+0800] {reddit_crawler.py:274} INFO - Connected to Reddit as: Key-Locksmith6742
[2025-04-13T21:58:43.253+0800] {reddit_crawler.py:120} INFO - Scraping submissions from the past 7 days (since 2025-04-06)
[2025-04-13T21:58:43.253+0800] {reddit_crawler.py:127} INFO - Scraping subreddit: r/travel
[2025-04-13T22:17:02.812+0800] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-04-13T22:17:02.818+0800] {process_utils.py:132} INFO - Sending 15 to group 62503. PIDs of all processes in the group: [62503]
[2025-04-13T22:17:02.818+0800] {process_utils.py:87} INFO - Sending the signal 15 to group 62503
[2025-04-13T22:17:02.819+0800] {taskinstance.py:3094} ERROR - Received SIGTERM. Terminating subprocesses.
[2025-04-13T22:17:02.829+0800] {taskinstance.py:3095} ERROR - Stacktrace: 
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/bin/airflow", line 8, in <module>
    sys.exit(main())
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/__main__.py", line 62, in main
    args.func(args)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 254, in _run_task_by_selected_method
    return _run_task_by_local_task_job(args, ti)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 322, in _run_task_by_local_task_job
    ret = run_job(job=job_runner.job, execute_callable=job_runner._execute)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/jobs/job.py", line 421, in run_job
    return execute_job(job, execute_callable=execute_callable)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/jobs/job.py", line 450, in execute_job
    ret = execute_callable()
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/jobs/local_task_job_runner.py", line 171, in _execute
    self.task_runner.start()
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 55, in start
    self.process = self._start_by_fork()
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/rencongchia/Y3S2/BT4301/project/airflow/dags/reddit_dag.py", line 70, in extract_reddit_data
    # Run the scraper
  File "/usr/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
    self._run_once()
  File "/usr/lib/python3.12/asyncio/base_events.py", line 1949, in _run_once
    event_list = self._selector.select(timeout)
  File "/usr/lib/python3.12/selectors.py", line 468, in select
    fd_event_list = self._selector.poll(timeout, max_ev)
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3095, in signal_handler
    self.log.error("Stacktrace: \n%s", "".join(traceback.format_stack()))

[2025-04-13T22:17:02.838+0800] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rencongchia/Y3S2/BT4301/project/airflow/dags/reddit_dag.py", line 70, in extract_reddit_data
    # Run the scraper
                      
  File "/usr/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
    self._run_once()
  File "/usr/lib/python3.12/asyncio/base_events.py", line 1949, in _run_once
    event_list = self._selector.select(timeout)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/selectors.py", line 468, in select
    fd_event_list = self._selector.poll(timeout, max_ev)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rencongchia/Y3S2/BT4301/project/.venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3097, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2025-04-13T22:17:02.842+0800] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-04-13T22:17:02.842+0800] {logging_mixin.py:190} INFO - Task start:2025-04-13 13:58:41.952979+00:00 end:2025-04-13 14:17:02.841375+00:00 duration:1100.888396
[2025-04-13T22:17:02.842+0800] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): extract_reddit_data> dag:<DAG: reddit_etl_pipeline> dagrun:<DagRun reddit_etl_pipeline @ 2025-04-13 13:22:15.449641+00:00: manual__2025-04-13T13:22:15.449641+00:00, state:running, queued_at: 2025-04-13 13:22:15.472379+00:00. externally triggered: True>
[2025-04-13T22:17:02.842+0800] {logging_mixin.py:190} INFO - Failure caused by Task received SIGTERM signal
[2025-04-13T22:17:02.842+0800] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=reddit_etl_pipeline, task_id=extract_reddit_data, run_id=manual__2025-04-13T13:22:15.449641+00:00, execution_date=20250413T132215, start_date=20250413T135841, end_date=20250413T141702
[2025-04-13T22:17:02.871+0800] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-04-13T22:17:02.911+0800] {process_utils.py:80} INFO - Process psutil.Process(pid=62503, status='terminated', exitcode=2, started='21:58:41') (62503) terminated with exit code 2
[2025-04-13T22:17:09.046+0800] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-13T22:17:09.052+0800] {taskinstance.py:2604} INFO - Dependencies not met for <TaskInstance: reddit_etl_pipeline.extract_reddit_data manual__2025-04-13T13:22:15.449641+00:00 [failed]>, dependency 'Task Instance State' FAILED: Task is in the 'failed' state.
[2025-04-13T22:17:09.052+0800] {local_task_job_runner.py:166} INFO - Task is not able to be run
